src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/build/netcoreapp1.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design.targets

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <MvcRazorContentRoot Condition="'$(MvcRazorContentRoot)'==''">$(MSBuildProjectDirectory)</MvcRazorContentRoot>
+    <MvcRazorOutputPath Condition="'$(MvcRazorOutputPath)'==''">$(OutputPath)</MvcRazorOutputPath>
+  </PropertyGroup>
+  <Target Name="Precompile"
+    AfterTargets="Precompile-Build;Precompile-Publish">
+    <ItemGroup>
+      <ExecArgs Include="
+        --runtimeconfig $(OutputPath)$(MSBuildProjectName).runtimeconfig.json;
+        --depsfile $(OutputPath)$(MSBuildProjectName).deps.json;
+        $(MSBuildThisFileDirectory)Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design.dll;
+        $(MSBuildProjectDirectory);
+        --output-path=$(MvcRazorOutputPath);
+        --application-name=$(MSBuildProjectName);
+        --content-root=$(MvcRazorContentRoot);" />
+      <ExecArgs Condition="$(DelaySign) == true"
+        Include="--delay-sign" />
+    </ItemGroup>
+    <Exec Command="dotnet exec @(ExecArgs->'%(Identity)', ' ')" />
```

Comment:
We played around with converting it to a Task. Unfortunately `dotnet msbuild` (the one's in the 1.0.0-preview4-004233 SDK) uses the 1.0.1 shared runtime. Precompilation references Mvc which is compiled against CoreFx 4.4.0 packages i.e versions newer than the one that are in the shared runtime running the task. This fails at runtime with errors like

```C#
D:\temp\msbuild-test\srcD:\temp\msbuild-test\src\src.csproj(4,5): error MSB4018: The "TestTask" task failed unexpectedly.\r
D:\temp\msbuild-test\src\src.csproj(4,5): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'System.Collections.Immutable, Version=1.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.\r
D:\temp\msbuild-test\src\src.csproj(4,5): error MSB4018: File name: 'System.Collections.Immutable, Version=1.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'\r
D:\temp\msbuild-test\src\src.csproj(4,5): error MSB4018:    at Microsoft.Test.TestTask.Test()\r
D:\temp\msbuild-test\src\src.csproj(4,5): error MSB4018:    at Microsoft.Test.TestTask.Execute()\r
D:\temp\msbuild-test\src\src.csproj(4,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()\r
D:\temp\msbuild-test\src\src.csproj(4,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext()
```

Super simple repro here: https://github.com/pranavkm/msbuild-repro/tree/master/src. We'll follow up with the MSBuild team next week, but given that we're also going to be loading user code, I think we are going to run in to issues converting it into a task. The current approach avoids these.